yt/visualization/plot_container.py

```diff
@@ -894,8 +896,23 @@ def set_zlim(self, field, zmin, zmax, dynamic_range=None):
             zmin = zmax / dynamic_range.
 
         """
-        myzmin = zmin
-        myzmax = zmax
+
+        def _sanitize_units(z):
+            # convert dimensionful inputs to float
+            if isinstance(z, tuple):
+                z = YTQuantity(*z)
```

Comment:
Can we change this to be `ds.quan` instead?  That way we can get units that are specific to the dataset, like comoving, hubble constant, redshift, etc.

Reply:
excellent suggestion, done !